src/edge_op.cpp

```diff
@@ -132,30 +201,62 @@ void Manifold::Impl::CleanupTopology() {
   // verts. They must be removed before edge collapse.
   SplitPinchedVerts();
 
+  Vec<int> entries;
+  FlagStore s;
   while (1) {
     ZoneScopedN("DedupeEdge");
 
     const size_t nbEdges = halfedge_.size();
     size_t numFlagged = 0;
 
-    Vec<SortEntry> entries;
-    entries.reserve(nbEdges / 2);
-    for (size_t i = 0; i < nbEdges; ++i) {
-      if (halfedge_[i].IsForward()) {
-        entries.push_back({halfedge_[i].startVert, halfedge_[i].endVert, i});
+#if MANIFOLD_PAR == 1
+    if (nbEdges > 1e5) {
+      entries.resize_nofill(nbEdges);
+      sequence(entries.begin(), entries.end());
+      stable_sort(entries.begin(), entries.end(), [&](int a, int b) {
+        const auto& self = halfedge_[a];
+        const auto& other = halfedge_[b];
+        // place all backward edges at the end
+        if (!other.IsForward()) return true;
+        if (!self.IsForward()) return false;
+        // dictionary order based on start and end vertices
```

Comment:
Why do we sort differently for parallel and serial?

Reply:
This is now explained in the comment:

```
      // Note that this is slightly different from the single thread version
      // because we store all indices instead of just indices of forward
      // halfedges. Backward halfedges are placed at the end by modifying the
      // comparison function.
```